packages/pulsar-updater/src/main.js

```diff
@@ -0,0 +1,213 @@
+            typeof objButtonForInstallMethod === "object" &&
+              objButtonForInstallMethod,
+          ],
+        }
+      );
+    } // else don't update, rely on cache set above
+  }
+
+  async newestRelease() {
+    superagent ??= require("superagent");
+
+    let res = await superagent
+      .get("https://api.github.com/repos/pulsar-edit/pulsar/releases")
+      .set("Accept", "application/vnd.github+json")
+      .set("User-Agent", "Pulsar.Pulsar-Updater");
+
+    if (res.status !== 200) {
+      // Lie and say it's something that will never update
+      return "0.0.0";
+    }
+
+    // We can be fast and simply check if the top of the array is newer than our
+    // current version. Since the return is ordered
+    return res.body[0].tag_name;
+  }
+
+  getNotificationText(installMethod, latestVersion) {
+    let returnText = `Pulsar ${latestVersion} is available.\n`;
+
+    switch (installMethod.installMethod) {
+      case "Developer Mode":
+        returnText +=
+          "Since you're in developer mode, Pulsy trusts you know how to update. :)";
+        break;
+      case "Safe Mode":
+        returnText +=
+          "Declining update suggestion since Pulsar is in Safe Mode.";
+        break;
+      case "Spec Mode":
+        returnText +=
+          "Declining update suggestion since Pulsar is in Spec Mode.";
+        break;
+      case "Flatpak Installation":
+        returnText += "Install the latest version by running `flatpak update`.";
+        break;
+      case "Deb-Get Installation":
+        returnText +=
+          "Install the latest version by running `sudo deb-get update`.";
+        break;
+      case "Nix Installation":
+        // TODO find nix update command
+        returnText += "Install the latest version via Nix.";
+        break;
+      case "Home Brew Installation":
+        returnText +=
+          "Install the latest version by running `brew upgrade pulsar`.";
+        break;
+      case "winget Installation":
+        returnText +=
+          "Install the latest version by running `winget upgrade pulsar`.";
+        break;
+      case "Chocolatey Installation":
+        returnText +=
+          "Install the latest version by running `choco upgrade pulsar`.";
+        break;
+      case "User Installation":
+      case "Machine Installation":
+      case "Portable Installation":
+      case "Manual Installation":
+      default:
+        returnText +=
+          "Download the latest version from the Pulsar Website or GitHub.";
+        break;
+    }
+
+    return returnText;
+  }
+
+  getObjButtonForInstallMethod(installMethod) {
+    let returnObj = null;
+
+    const openWebGitHub = (e) => {
+      e.preventDefault();
+      shell = shell || require("electron").shell;
+      shell.openExternal(
+        `https://github.com/pulsar-edit/pulsar/releases/tag/${
+          this.cache.getCacheItem("last-update-check").latestVersion
+        }`
+      );
+    };
```

Comment:
```suggestion
    const openWebGitHub = (e) => {
      e.preventDefault();
      shell ??= shell || require("electron").shell;
      let latestVersion = this.cache.getCacheItem("last-update-check")?.latestVersion;
      let tagSegment = latestVersion ? `tag/${latestVersion}` : "";
      shell.openExternal(`https://github.com/pulsar-edit/pulsar/releases/${tagSegment}`);
    };
```

Haven't tested this snippet, but I had the instinct to try to guard against a failed cache lookup, even if it's implausible. If the lookup fails, we'll just bring the user to `https://github.com/pulsar-edit/pulsar/releases/`, which will have the latest release at the top anyway.

Reply:
Very smart move, yeah in it's current implementation the cache should always be set prior to this point, but in case theres change in the package later this is a really good idea.